portal/static/js/ajaxWorker.js

```diff
@@ -0,0 +1,70 @@
+var requestAttempts = 0;
+function sendRequest(url, params, callback) { //XHR request in pure JavaScript
```

Comment:
Still learning about service workers, but is there a reason you can't use the jQuery AJAX wrappers?

Reply:
@ivan-c  ivan, i think that one of the limitations that a web worker has is that it will only have access to a subset of javascript features (e.g. XMLHTTP Request object, navigation object, Application Cache).  If you try accessing Jquery related features, you will get runtime errors.  And you are right, it won't be able to manipulate DOM.  so all this worker does is just returning data from the XMLHttp request but running this on a separate background thread - this, I hope, will boost the performance. I  was reading up on this before deciding to trying it out:  https://www.html5rocks.com/en/tutorials/workers/basics

Reply:
Thanks for the background, I'll have to read more!